lib/Echidna/Output/Corpus.hs

```diff
@@ -26,3 +31,35 @@ loadTxs dir = do
   putStrLn ("Loaded " ++ show (length txSeqs) ++ " transaction sequences from " ++ dir)
   pure txSeqs
   where readCall f = decodeStrict <$> BS.readFile f
+
+-- save to corpus in the background while tests are running
+-- returns a channel that gets sent () when the process is done
+runCorpusSaver :: Env -> IO (Chan ())
+runCorpusSaver env = case env.cfg.campaignConf.corpusDir of
+  Nothing -> do
+    finishChan <- newChan
+    writeChan finishChan ()
+    pure finishChan
+  Just dir -> do
+    -- we want to dupChan *before* forking so we don't miss any events
+    chan <- dupChan env.eventQueue
+    finishChan <- newChan
+    void $ forkFinally (loop dir chan nworkers) (const $ writeChan finishChan ())
+    pure finishChan
+  where
+    nworkers :: Int
+    nworkers = fromIntegral $ fromMaybe 1 env.cfg.campaignConf.workers
+
+    loop !dir !chan !workersAlive = when (workersAlive > 0) $ do
```

Comment:
we have some logic duplication here but we can refactor this with the subscribe system we talked about

Reply:
Separate PR for that subscribe system? I could add it to this PR if you want

Reply:
let's do it in this PR then